tests/kuttl/common/assert_keystone_deployment.yaml

```diff
@@ -0,0 +1,134 @@
+#
+# Check for:
+#
+# - 1 KeystoneAPI CR
+# - 1 Pod for KeystoneAPI CR
+#
+
+apiVersion: keystone.openstack.org/v1beta1
+kind: KeystoneAPI
+metadata:
+  finalizers:
+  - KeystoneAPI
+  name: keystone
+  namespace: openstack
+spec:
+  adminProject: admin
+  adminRole: admin
+  adminUser: admin
+  containerImage: quay.io/tripleozedcentos9/openstack-keystone:current-tripleo
+  customServiceConfig: |
+    [DEFAULT]
+    debug = true
+  databaseInstance: openstack
+  databaseUser: keystone
+  debug:
+    bootstrap: false
+    dbSync: false
+    service: false
+  passwordSelectors:
+    admin: AdminPassword
+    database: KeystoneDatabasePassword
+  preserveJobs: false
+  region: regionOne
+  replicas: 1
+  resources:
+    requests:
+      cpu: "1.0"
+      memory: 500Mi
+  secret: osp-secret
+status:
+  apiEndpoint:
+    admin: http://keystone-admin-openstack.apps-crc.testing
+    internal: http://keystone-internal-openstack.apps-crc.testing
+    public: http://keystone-public-openstack.apps-crc.testing
+  databaseHostname: openstack
+  readyCount: 1
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    openshift.io/scc: anyuid
+  labels:
+    service: keystone
+  #name: keystone-69f88774bd-gjspj   Can't include name due to random append
```

Comment:
the operators created the deployment, not the pod. should we instead check if the deployment and its pod spec in there is correct and the readycount for the pods?

Reply:
Done. There is only one field that can't be obtained from the deployment, the `openshift.io/scc: anyuid` annotation that is only obtainable from the pod. I'm not sure how relevant that is for us, so I left a check for it, but can be removed if it's not that relevant.